magpie/api/management/resource/resource_utils.py

```diff
@@ -115,7 +115,7 @@ def get_resource_permissions(resource, db_session):
     ax.verify_param(resource, notNone=True, httpError=HTTPBadRequest, paramName=u"resource",
                     msgOnFail=s.UserResourcePermissions_GET_BadRequestResourceResponseSchema.description)
     # directly access the service resource
-    if resource.root_service_id is None:
+    if resource.root_service_id is None or resource.root_service_id == resource.resource_id:
```

Comment:
Was this case encountered somewhere?
`resource.root_service_id` is expected to always be `None` if it is a service.
If this is not respected, it's an implementation error somewhere else that must be fixed as it could break other code logic.

Reply:
Hum... I think I might have to do a small database migration here. In my last PR, I fixed a database migration, but I thought that setting the `root_service_id` to the `resource_id` for the services was the intention. (see `a395ef9d3fe6_reference_root_service.py`)

I could add the migration in this PR and remove this check.

Reply:
Ok, yes please adjust within this PR.

Reply:
done in https://github.com/Ouranosinc/Magpie/pull/228/commits/f6d7700f567bed88201db57c9f0a21e4913b3b9a